cachito/workers/pkg_managers/rubygems.py

```diff
@@ -312,11 +322,114 @@ def _download_git_package(gem, rubygems_deps_dir, rubygems_raw_repo_name, nexus_
         # Copy downloaded archive to expected download path
         shutil.copy(repo_name.sources_dir.archive_path, download_path)
 
+    url = gem.source
+    ref = gem.version.lower()
+
     return {
-        "package": gem.name,
+        "name": gem.name,
+        "version": f"git+{url}@{ref}",
         "path": download_path,
-        "url": gem.source,
-        "ref": gem.version.lower(),
         "raw_component_name": raw_component_name,
         "have_raw_component": have_raw_component,
     }
+
+
+def _get_path_package_info(dep, bundle_dir, package_root):
+    """
+    Get info about PATH dependency including path relative to the bundle source root.
+
+    :param GemMetadata dep: path dependency
+    :param bundle_dir: the root of the bundle with app source code
+    :param package_root: path to the root of the processed package
+    :return: dict with name, version and path keys
+    """
+    path = Path(package_root / dep.source).resolve().relative_to(bundle_dir.source_root_dir)
+
+    return {
+        "name": dep.name,
+        "version": dep.version,
+        "path": path,
+    }
+
+
+def resolve_rubygems(package_root, request):
+    """
+    Resolve and fetch RubyGems dependencies for the given app source archive.
+
+    :param Path package_root: the full path to the package root
+    :param dict request: the Cachito request to resolve RubyGems dependencies for
+    :return: a dictionary that has the following keys:
+        ``dependencies`` which is a list of dicts representing the package Dependencies
+        ``gemfile_lock`` an absolute path to the Gemfile.lock
+    """
+    gemlock_path = package_root / GEMFILE_LOCK
+    dependencies = parse_gemlock(package_root, gemlock_path)
+    dependencies = download_dependencies(request["id"], dependencies, package_root)
+
+    rubygems_repo_name = get_rubygems_hosted_name(request["id"])
+    for dependency in dependencies:
+        if dependency["kind"] == "GEM":
+            _push_downloaded_gem(dependency, rubygems_repo_name)
+
+    return {
+        "package": {"type": "rubygems"},
+        "dependencies": dependencies,
+        "gemfile_lock": gemlock_path,
```

Comment:
Will we need "gemfile_lock" for something? AFAIK pip only uses the paths when rewriting requirements files to point to git and https deps to the Nexus repo

Reply:
Yes, let's say there are two packages in the repository, `package1` and `package2`. We want to be able to run `bundle install` from both `repo/package1` and `repo/package2` directories. In order to redirect Bundler to use Nexus or local git repos, we have to create config files at `repo/package1/.bundle/config` and similarly for second package. And to be able to properly determine the path where to create these config files, we would do `config_file_path = gemlock_path.parent / Path(".bundle/config")`. Note that there's no support of alternative locations for `Gemfile.lock`, so we expect it always to be `package/Gemfile.lock`. Clearly, we could replace this with something like package to the package.

Reply:
Does the function need to return the path? Doesn't the caller of the `<generate bundle config>` function already know the path?

Reply:
Yes, you're right.